doc/BMI_MODELS.md

```diff
@@ -259,6 +271,48 @@ Similarly, different compilers (or different compiler versions) may implement `d
 
 An example implementation for an appropriate BMI model as a **C++** shared library is provided in the project [here](../extern/test_bmi_cpp).
 
+## BMI Models Written in Python
+
+  - [Enabling Python Integration](#enabling-python-integration)
+  - [BMI Python Model as Package Class](#bmi-python-model-as-package-class)
+  - [BMI Python Example](#bmi-python-example)
+
+### Enabling Python Integration
+
+Python integration is controlled with the CMake build flag `NGEN_ACTIVATE_PYTHON`, however this currently defaults to "On"--you would need to turn this off if Python is not available in your environment. See [the Dependencies documentation](DEPENDENCIES.md#python-3-libraries) for specifics on Python requirements, but in summary you will need a working Python environment with NumPy installed. You can set up a Python environment anywhere with the usual environment variables, but note that ngen will look for a Python environment in the build directory named `.venv` or `venv` by default. The appropriate Python environment should be active in the shell when ngen is run.
+
+For Python BMI models specifically, you will also need to install the [bmipy](https://github.com/csdms/bmi-python) package, which provides a base class for Python BMI models.
+
+### BMI Python Model as Package Class
+
+To use a Python BMI model, the model needs to be installed as a package in the Python environment and the package must have a class that extends bmipy, like so
+
+```python
+from bmipy import Bmi
+class bmi_model(Bmi):
+    ...
+```
+
+**TIP:** If you are actively developing a Python BMI model, you may want to [install your package with the `-e` flag](https://pip.pypa.io/en/stable/topics/local-project-installs/#editable-installs).
```

Comment:
Is it worth going into making the class available by appending to `PYTHONPATH`?  Using `pip -e` is probably a better solution in general, but, e.g., assumes a user  can install packages in the environment.

Reply:
If they can't they are going to have bigger problems, probably... they can't get `bmipy` for instance or likely a working version of NumPy. I would leave it out... there's only so much we can do here. Don't get me started on all the various `setup.cfg` or `manifest.json` or whatever you're supposed to do this week to make a Python package that we could go into but won't.